cpp/src/arrow/python/udf.h

```diff
@@ -41,6 +41,8 @@ struct ARROW_PYTHON_EXPORT ScalarUdfOptions {
   std::shared_ptr<DataType> output_type;
 };
 
+/// \brief A context defined to hold meta-data required in
+/// scalar UDF execution.
```

Comment:
It's fine to add this docstring, but by API documentation here we mean *Python* API (i.e. functions and classes exposed in the `pyarrow.compute` namespace).

Reply:
Yes, I think all the docs are within the `_compute.pyx` file but not in the `compute.py` file because we just referencing it in the imports. Would that avoid the API docs being generated? This part I am not quite sure.

Reply:
Any suggestions?